### PR TITLE
[fix](multi-catalog) add config to disable external DDL

### PR DESCRIPTION
**Pull request by wsjz**
## Proposed changes

add config to disable external DDL

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by morningman — DISMISSED** [review]

**Comment by morningman** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]

**Comment by wsjz** [comment]
image
effect internal table when use external catalog

**Comment by wsjz** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java [hunk 1]
```diff
@@ -3042,6 +3042,12 @@ public void replayRenameDatabase(String dbName, String newDbName) {
      */
     public void createTable(CreateTableStmt stmt) throws UserException {
         getCurrentCatalog().createTable(stmt);
+        CatalogIf<?> catalogIf = catalogMgr.getCatalog(stmt.getCatalogName());
+        if (catalogIf == null) {
+            getCurrentCatalog().createTable(stmt);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java:3047]: We should throw exception if catalog does not exist

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java [hunk 2]
```diff
@@ -3671,7 +3677,12 @@ public void replayAlterExternalTableSchema(String dbName, String tableName, List
 
     // Drop table
     public void dropTable(DropTableStmt stmt) throws DdlException {
-        getCurrentCatalog().dropTable(stmt);
+        CatalogIf<?> catalogIf = catalogMgr.getCatalog(stmt.getCatalogName());
+        if (catalogIf == null) {
+            getCurrentCatalog().dropTable(stmt);
+        } else {
+            catalogIf.dropTable(stmt);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java:3684]: ditto

**Comment by wsjz** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by wuwenchi — APPROVED** [review]

**Pull request merged**